### PR TITLE
Fix some typos

### PR DESCRIPTION
**Pull request by lucasmerlin**
The spell check pipeline in #5313 suddenly failed, this fixes these typos and some more found via my IDEs spell checker tool

### CODE REVIEW

File: crates/egui/src/util/cache.rs [hunk 1]
```diff
@@ -39,7 +39,7 @@ impl<Value, Computer> FrameCache<Value, Computer> {
     }
 
     /// Must be called once per frame to clear the cache.
-    pub fn evice_cache(&mut self) {
+    pub fn evict_cache(&mut self) {
         let current_generation = self.generation;
         self.cache.retain(|_key, cached| {
             cached.0 == current_generation // only keep those that were used this frame
```
Comment (lucasmerlin) [inline crates/egui/src/util/cache.rs:45]: This is a breaking change but as far as I can tell this is only called in CacheTrait::update and no external code calls this ([confirmed via github code search](https://github.com/search?q=%2F%5CWevice_cache%2F&type=code))

File: crates/egui/src/widgets/checkbox.rs [hunk 2]
```diff
@@ -38,7 +38,7 @@ impl<'a> Checkbox<'a> {
 
     /// Display an indeterminate state (neither checked nor unchecked)
     ///
-    /// This only affects the checkbox's appearance. It will still toggle its boolean value when
+    /// This only affects the checkboxes appearance. It will still toggle its boolean value when
```
Comment (YgorSouza) [inline crates/egui/src/widgets/checkbox.rs:41]: This one isn't a typo.

File: crates/epaint/src/tessellator.rs [hunk 3]
```diff
@@ -2029,8 +2029,8 @@ impl Tessellator {
         use rayon::prelude::*;
 
         // We only parallelize large/slow stuff, because each tessellation job
-        // will allocate a new Mesh, and so it creates a lot of extra memory framentation
-        // and callocations that is only worth it for large shapes.
+        // will allocate a new Mesh, and so it creates a lot of extra memory fragmentation
+        // and call locations that is only worth it for large shapes.
```
Comment (YgorSouza) [inline crates/epaint/src/tessellator.rs:2033]: I think this is supposed to be "allocations".

File: crates/egui/src/widgets/checkbox.rs (hunk 2 above)
Comment (lucasmerlin) [reply crates/egui/src/widgets/checkbox.rs:41]: Huh, the grammar check marked it as wrong but I guess the original version was correct. I'll revert it

File: crates/epaint/src/tessellator.rs (hunk 3 above)
Comment (lucasmerlin) [reply crates/epaint/src/tessellator.rs:2033]: Ah, that makes much more sense

**Pull request merged**